fv3core/stencils/remap_profile.py

```diff
@@ -469,158 +467,207 @@ def set_inner(
+        )
+        self._q_bot: FloatField = utils.make_storage_from_shape(
+            grid.domain_shape_full(add=(0, 0, 1)), origin=self._full_orig
+        )
+        self._extm: FloatField = utils.make_storage_from_shape(
+            grid.domain_shape_full(add=(0, 0, 1)), origin=self._full_orig
+        )
+        self._ext5: FloatField = utils.make_storage_from_shape(
+            grid.domain_shape_full(add=(0, 0, 1)), origin=self._full_orig
+        )
+        self._ext6: FloatField = utils.make_storage_from_shape(
+            grid.domain_shape_full(add=(0, 0, 1)), origin=self._full_orig
         )
 
-        set_top(
-            a4_1,
-            a4_2,
-            a4_3,
-            a4_4,
-            extm,
-            iv,
-            origin=orig,
-            domain=(i_extent, j_extent, 2),
+        self._set_values_stencil = gtscript.stencil(
+            definition=set_vals,
+            externals={"iv": iv, "kord": abs(kord)},
+            backend=global_config.get_backend(),
+            rebuild=global_config.get_rebuild(),
         )
 
-        set_inner(
-            a4_1,
-            a4_2,
-            a4_3,
-            a4_4,
-            gam,
-            extm,
-            ext5,
-            ext6,
-            qmin,
-            abs(kord),
-            iv,
-            origin=(i1, j1, 2),
-            domain=(i_extent, j_extent, km - 4),
+        self._apply_constraints_stencil = gtscript.stencil(
+            definition=apply_constraints,
+            externals={"iv": iv, "kord": abs(kord)},
+            backend=global_config.get_backend(),
+            rebuild=global_config.get_rebuild(),
+        )
+
+        self._set_top_stencil = gtscript.stencil(
+            definition=set_top,
+            externals={"iv": iv},
+            backend=global_config.get_backend(),
+            rebuild=global_config.get_rebuild(),
+        )
+
+        self._set_set_inner_stencil = gtscript.stencil(
+            definition=set_inner,
+            externals={"iv": iv, "kord": abs(kord)},
+            backend=global_config.get_backend(),
+            rebuild=global_config.get_rebuild(),
+        )
+
+        self._set_bottom_stencil = gtscript.stencil(
+            definition=set_bottom,
+            externals={"iv": iv},
+            backend=global_config.get_backend(),
+            rebuild=global_config.get_rebuild(),
         )
 
-        set_bottom(
+    def __call__(
+        self,
+        qs: FloatField,
+        a4_1: FloatField,
+        a4_2: FloatField,
+        a4_3: FloatField,
+        a4_4: FloatField,
+        delp: FloatField,
+        i1: int,
+        i2: int,
+        j1: int,
+        j2: int,
+        qmin: float = 0.0,
+    ):
+        """
+        Calculates the interpolation coefficients for a cubic-spline which models the
+        distribution of the remapped field within each deformed grid cell.
+        The constraints on the spline are set by kord and iv.
```

Comment:
excellent docstring, you could potentially go into more detail about what kord and iv are, but don't have to for this pr